The examples no longer work

Good afternoon

### Description

Some time ago I have made available on GitHub a package of parallel manipulators modelling:

https://github.com/looms-polimi/Parallel_manipulators

where all examples worked.
I have downloaded the package but now the examples do not not work.

### Steps to Reproduce

Open the package.mo file in 

.\Parallel_manipulators-master\Parallel_manipulators-master\Modelica_models\StewartPlatform

and run StewartPlatform.Scenarios.StewartPlatform_InverseDynamic3_noBus
Simply check the model and got:

[1] 16:09:05 Translation Error
[C:/OM120/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFType.mo: 1112:13-1113:80]: Internal error NFType.subscript got unsubscriptable type StewartPlatform.Interfaces.Pose

### Expected Behavior

Wen the model run, plotting the outputs of TorqueComparison should give the following picture:

### Screenshots

![image](https://user-images.githubusercontent.com/17689020/213873496-217d671c-e3bc-42fe-89c5-d1e6d5ad8b80.png)

Moreover, the following animation is available:

![image](https://user-images.githubusercontent.com/17689020/213873702-82a33a51-553e-47e4-b2c0-518e69dead5b.png)


### Version and OS

<!--- Please complete the following information. -->

 - OpenModelica Version: [e.g. `omc --version` or `Help->About OMEdit` from OMEdit]

Connected to OpenModelica v1.20.0 (64-bit)
Connected to OMSimulator v2.1.1.post188-gaf996ad-mingw

 - OS: [e.g. Windows 10, 64 bit]

Windows 11 64 bit

 - Versions of used Modelica libraries if applicable

Modelica 4.0.0
Version Date: 2020-06-04

### Additional Context

<!--- Add any other context about the problem here. -->

Yes, I got this

C:/Program Files/OpenModelica1.21.0-dev-64bit/share/omc/scripts/Compile.bat StewartPlatform_InverseDynamic3_noBus gcc mingw64 parallel dynamic 8 0
PATH = "C:\PROGRA~1\OPENMO~1.0-D\tools\msys\mingw64\bin;C:\PROGRA~1\OPENMO~1.0-D\tools\msys\mingw64\bin\..\..\usr\bin;"
mingw32-make: Entering directory 'C:/Users/jackj/AppData/Local/Temp/OPENMO~1/OMEdit/STEWAR~1.STE'
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus.o StewartPlatform_InverseDynamic3_noBus.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_functions.o StewartPlatform_InverseDynamic3_noBus_functions.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_records.o StewartPlatform_InverseDynamic3_noBus_records.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_01exo.o StewartPlatform_InverseDynamic3_noBus_01exo.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_02nls.o StewartPlatform_InverseDynamic3_noBus_02nls.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_03lsy.o StewartPlatform_InverseDynamic3_noBus_03lsy.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_04set.o StewartPlatform_InverseDynamic3_noBus_04set.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_05evt.o StewartPlatform_InverseDynamic3_noBus_05evt.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_06inz.o StewartPlatform_InverseDynamic3_noBus_06inz.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_07dly.o StewartPlatform_InverseDynamic3_noBus_07dly.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_08bnd.o StewartPlatform_InverseDynamic3_noBus_08bnd.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_09alg.o StewartPlatform_InverseDynamic3_noBus_09alg.c
StewartPlatform_InverseDynamic3_noBus.c:10345:606: error: too many arguments to function call, expected 6, have 9
  real_array_copy_data(omc_StewartPlatform_Functions_legsLength(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1463] /* controller.inverseKinematic.base.D PARAM */), (data->simulationInfo->realParameter[1464] /* controller.inverseKinematic.base.De PARAM */), (data->simulationInfo->realParameter[1469] /* controller.inverseKinematic.base.thickness PARAM */), (data->simulationInfo->realParameter[1468] /* controller.inverseKinematic.base.mass PARAM */), (data->simulationInfo->realParameter[1465] /* controller.inverseKinematic.base.alpha PARAM */), (data->simulationInfo->realParameter[1466] /* controller.inverseKinematic.base.area PARAM */), (data->simulationInfo->realParameter[1470] /* controller.inverseKinematic.base.volume PARAM */), (data->simulationInfo->realParameter[1467] /* controller.inverseKinematic.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1489] /* controller.inverseKinematic.platform.D PARAM */), (data->simulationInfo->realParameter[1490] /* controller.inverseKinematic.platform.De PARAM */), (data->simulationInfo->realParameter[1495] /* controller.inverseKinematic.platform.thickness PARAM */), (data->simulationInfo->realParameter[1494] /* controller.inverseKinematic.platform.mass PARAM */), (data->simulationInfo->realParameter[1491] /* controller.inverseKinematic.platform.alpha PARAM */), (data->simulationInfo->realParameter[1492] /* controller.inverseKinematic.platform.area PARAM */), (data->simulationInfo->realParameter[1496] /* controller.inverseKinematic.platform.volume PARAM */), (data->simulationInfo->realParameter[1493] /* controller.inverseKinematic.platform.density PARAM */)), tmp920, tmp921, tmp922), tmp923);
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c/util/generic_array.h:48:76: note: expanded from macro 'real_array_copy_data'
#define real_array_copy_data(src,dst)               simple_array_copy_data(src, &dst, sizeof(modelica_real));
                                                                           ^~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
StewartPlatform_InverseDynamic3_noBus.c:10345:1446: error: too many arguments to function call, expected 6, have 9
  real_array_copy_data(omc_StewartPlatform_Functions_legsLength(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1463] /* controller.inverseKinematic.base.D PARAM */), (data->simulationInfo->realParameter[1464] /* controller.inverseKinematic.base.De PARAM */), (data->simulationInfo->realParameter[1469] /* controller.inverseKinematic.base.thickness PARAM */), (data->simulationInfo->realParameter[1468] /* controller.inverseKinematic.base.mass PARAM */), (data->simulationInfo->realParameter[1465] /* controller.inverseKinematic.base.alpha PARAM */), (data->simulationInfo->realParameter[1466] /* controller.inverseKinematic.base.area PARAM */), (data->simulationInfo->realParameter[1470] /* controller.inverseKinematic.base.volume PARAM */), (data->simulationInfo->realParameter[1467] /* controller.inverseKinematic.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1489] /* controller.inverseKinematic.platform.D PARAM */), (data->simulationInfo->realParameter[1490] /* controller.inverseKinematic.platform.De PARAM */), (data->simulationInfo->realParameter[1495] /* controller.inverseKinematic.platform.thickness PARAM */), (data->simulationInfo->realParameter[1494] /* controller.inverseKinematic.platform.mass PARAM */), (data->simulationInfo->realParameter[1491] /* controller.inverseKinematic.platform.alpha PARAM */), (data->simulationInfo->realParameter[1492] /* controller.inverseKinematic.platform.area PARAM */), (data->simulationInfo->realParameter[1496] /* controller.inverseKinematic.platform.volume PARAM */), (data->simulationInfo->realParameter[1493] /* controller.inverseKinematic.platform.density PARAM */)), tmp920, tmp921, tmp922), tmp923);
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c/util/generic_array.h:48:76: note: expanded from macro 'real_array_copy_data'
#define real_array_copy_data(src,dst)               simple_array_copy_data(src, &dst, sizeof(modelica_real));
                                                                           ^~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_10asr.o StewartPlatform_InverseDynamic3_noBus_10asr.c
StewartPlatform_InverseDynamic3_noBus.c:33917:465: error: too many arguments to function call, expected 6, have 9
    if(!omc_StewartPlatform_Functions_validPose(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1620] /* gp.base.D PARAM */), (data->simulationInfo->realParameter[1621] /* gp.base.De PARAM */), (data->simulationInfo->realParameter[1626] /* gp.base.thickness PARAM */), (data->simulationInfo->realParameter[1625] /* gp.base.mass PARAM */), (data->simulationInfo->realParameter[1622] /* gp.base.alpha PARAM */), (data->simulationInfo->realParameter[1623] /* gp.base.area PARAM */), (data->simulationInfo->realParameter[1627] /* gp.base.volume PARAM */), (data->simulationInfo->realParameter[1624] /* gp.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1762] /* gp.platform.D PARAM */), (data->simulationInfo->realParameter[1763] /* gp.platform.De PARAM */), (data->simulationInfo->realParameter[1768] /* gp.platform.thickness PARAM */), (data->simulationInfo->realParameter[1767] /* gp.platform.mass PARAM */), (data->simulationInfo->realParameter[1764] /* gp.platform.alpha PARAM */), (data->simulationInfo->realParameter[1765] /* gp.platform.area PARAM */), (data->simulationInfo->realParameter[1769] /* gp.platform.volume PARAM */), (data->simulationInfo->realParameter[1766] /* gp.platform.density PARAM */)), tmp105, tmp106, tmp107, 1.798, 0.998))
                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
StewartPlatform_InverseDynamic3_noBus.c:33917:1105: error: too many arguments to function call, expected 6, have 9
    if(!omc_StewartPlatform_Functions_validPose(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1620] /* gp.base.D PARAM */), (data->simulationInfo->realParameter[1621] /* gp.base.De PARAM */), (data->simulationInfo->realParameter[1626] /* gp.base.thickness PARAM */), (data->simulationInfo->realParameter[1625] /* gp.base.mass PARAM */), (data->simulationInfo->realParameter[1622] /* gp.base.alpha PARAM */), (data->simulationInfo->realParameter[1623] /* gp.base.area PARAM */), (data->simulationInfo->realParameter[1627] /* gp.base.volume PARAM */), (data->simulationInfo->realParameter[1624] /* gp.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1762] /* gp.platform.D PARAM */), (data->simulationInfo->realParameter[1763] /* gp.platform.De PARAM */), (data->simulationInfo->realParameter[1768] /* gp.platform.thickness PARAM */), (data->simulationInfo->realParameter[1767] /* gp.platform.mass PARAM */), (data->simulationInfo->realParameter[1764] /* gp.platform.alpha PARAM */), (data->simulationInfo->realParameter[1765] /* gp.platform.area PARAM */), (data->simulationInfo->realParameter[1769] /* gp.platform.volume PARAM */), (data->simulationInfo->realParameter[1766] /* gp.platform.density PARAM */)), tmp105, tmp106, tmp107, 1.798, 0.998))
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_11mix.o StewartPlatform_InverseDynamic3_noBus_11mix.c
4 errors generated.
mingw32-make: *** [<builtin>: StewartPlatform_InverseDynamic3_noBus.o] Error 1
mingw32-make: *** Waiting for unfinished jobs....
mingw32-make: Leaving directory 'C:/Users/jackj/AppData/Local/Temp/OPENMO~1/OMEdit/STEWAR~1.STE'
Compilation process failed. Exited with code 0x00000002.

Good afternoon,

Installed today's nightly build but the error still remain:

C:/Program Files/OpenModelica1.21.0-dev-64bit/share/omc/scripts/Compile.bat StewartPlatform_InverseDynamic3_noBus gcc mingw64 parallel dynamic 8 0
PATH = "C:\PROGRA~1\OPENMO~1.0-D\tools\msys\mingw64\bin;C:\PROGRA~1\OPENMO~1.0-D\tools\msys\mingw64\bin\..\..\usr\bin;"
mingw32-make: Entering directory 'C:/Users/jackj/AppData/Local/Temp/OPENMO~1/OMEdit/STEWAR~1.STE'
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus.o StewartPlatform_InverseDynamic3_noBus.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_functions.o StewartPlatform_InverseDynamic3_noBus_functions.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_records.o StewartPlatform_InverseDynamic3_noBus_records.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_01exo.o StewartPlatform_InverseDynamic3_noBus_01exo.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_02nls.o StewartPlatform_InverseDynamic3_noBus_02nls.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_03lsy.o StewartPlatform_InverseDynamic3_noBus_03lsy.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_04set.o StewartPlatform_InverseDynamic3_noBus_04set.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_05evt.o StewartPlatform_InverseDynamic3_noBus_05evt.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_06inz.o StewartPlatform_InverseDynamic3_noBus_06inz.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_07dly.o StewartPlatform_InverseDynamic3_noBus_07dly.c
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_08bnd.o StewartPlatform_InverseDynamic3_noBus_08bnd.c
StewartPlatform_InverseDynamic3_noBus.c:10345:606: error: too many arguments to function call, expected 6, have 9
  real_array_copy_data(omc_StewartPlatform_Functions_legsLength(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1463] /* controller.inverseKinematic.base.D PARAM */), (data->simulationInfo->realParameter[1464] /* controller.inverseKinematic.base.De PARAM */), (data->simulationInfo->realParameter[1469] /* controller.inverseKinematic.base.thickness PARAM */), (data->simulationInfo->realParameter[1468] /* controller.inverseKinematic.base.mass PARAM */), (data->simulationInfo->realParameter[1465] /* controller.inverseKinematic.base.alpha PARAM */), (data->simulationInfo->realParameter[1466] /* controller.inverseKinematic.base.area PARAM */), (data->simulationInfo->realParameter[1470] /* controller.inverseKinematic.base.volume PARAM */), (data->simulationInfo->realParameter[1467] /* controller.inverseKinematic.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1489] /* controller.inverseKinematic.platform.D PARAM */), (data->simulationInfo->realParameter[1490] /* controller.inverseKinematic.platform.De PARAM */), (data->simulationInfo->realParameter[1495] /* controller.inverseKinematic.platform.thickness PARAM */), (data->simulationInfo->realParameter[1494] /* controller.inverseKinematic.platform.mass PARAM */), (data->simulationInfo->realParameter[1491] /* controller.inverseKinematic.platform.alpha PARAM */), (data->simulationInfo->realParameter[1492] /* controller.inverseKinematic.platform.area PARAM */), (data->simulationInfo->realParameter[1496] /* controller.inverseKinematic.platform.volume PARAM */), (data->simulationInfo->realParameter[1493] /* controller.inverseKinematic.platform.density PARAM */)), tmp920, tmp921, tmp922), tmp923);
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c/util/generic_array.h:48:76: note: expanded from macro 'real_array_copy_data'
#define real_array_copy_data(src,dst)               simple_array_copy_data(src, &dst, sizeof(modelica_real));
                                                                           ^~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
StewartPlatform_InverseDynamic3_noBus.c:10345:1446: error: too many arguments to function call, expected 6, have 9
  real_array_copy_data(omc_StewartPlatform_Functions_legsLength(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1463] /* controller.inverseKinematic.base.D PARAM */), (data->simulationInfo->realParameter[1464] /* controller.inverseKinematic.base.De PARAM */), (data->simulationInfo->realParameter[1469] /* controller.inverseKinematic.base.thickness PARAM */), (data->simulationInfo->realParameter[1468] /* controller.inverseKinematic.base.mass PARAM */), (data->simulationInfo->realParameter[1465] /* controller.inverseKinematic.base.alpha PARAM */), (data->simulationInfo->realParameter[1466] /* controller.inverseKinematic.base.area PARAM */), (data->simulationInfo->realParameter[1470] /* controller.inverseKinematic.base.volume PARAM */), (data->simulationInfo->realParameter[1467] /* controller.inverseKinematic.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1489] /* controller.inverseKinematic.platform.D PARAM */), (data->simulationInfo->realParameter[1490] /* controller.inverseKinematic.platform.De PARAM */), (data->simulationInfo->realParameter[1495] /* controller.inverseKinematic.platform.thickness PARAM */), (data->simulationInfo->realParameter[1494] /* controller.inverseKinematic.platform.mass PARAM */), (data->simulationInfo->realParameter[1491] /* controller.inverseKinematic.platform.alpha PARAM */), (data->simulationInfo->realParameter[1492] /* controller.inverseKinematic.platform.area PARAM */), (data->simulationInfo->realParameter[1496] /* controller.inverseKinematic.platform.volume PARAM */), (data->simulationInfo->realParameter[1493] /* controller.inverseKinematic.platform.density PARAM */)), tmp920, tmp921, tmp922), tmp923);
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c/util/generic_array.h:48:76: note: expanded from macro 'real_array_copy_data'
#define real_array_copy_data(src,dst)               simple_array_copy_data(src, &dst, sizeof(modelica_real));
                                                                           ^~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
clang  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.21.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=StewartPlatform_InverseDynamic3_noBus -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=99 -DOMC_NUM_NONLINEAR_SYSTEMS=6 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o StewartPlatform_InverseDynamic3_noBus_09alg.o StewartPlatform_InverseDynamic3_noBus_09alg.c
StewartPlatform_InverseDynamic3_noBus.c:33917:465: error: too many arguments to function call, expected 6, have 9
    if(!omc_StewartPlatform_Functions_validPose(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1620] /* gp.base.D PARAM */), (data->simulationInfo->realParameter[1621] /* gp.base.De PARAM */), (data->simulationInfo->realParameter[1626] /* gp.base.thickness PARAM */), (data->simulationInfo->realParameter[1625] /* gp.base.mass PARAM */), (data->simulationInfo->realParameter[1622] /* gp.base.alpha PARAM */), (data->simulationInfo->realParameter[1623] /* gp.base.area PARAM */), (data->simulationInfo->realParameter[1627] /* gp.base.volume PARAM */), (data->simulationInfo->realParameter[1624] /* gp.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1762] /* gp.platform.D PARAM */), (data->simulationInfo->realParameter[1763] /* gp.platform.De PARAM */), (data->simulationInfo->realParameter[1768] /* gp.platform.thickness PARAM */), (data->simulationInfo->realParameter[1767] /* gp.platform.mass PARAM */), (data->simulationInfo->realParameter[1764] /* gp.platform.alpha PARAM */), (data->simulationInfo->realParameter[1765] /* gp.platform.area PARAM */), (data->simulationInfo->realParameter[1769] /* gp.platform.volume PARAM */), (data->simulationInfo->realParameter[1766] /* gp.platform.density PARAM */)), tmp105, tmp106, tmp107, 1.798, 0.998))
                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
StewartPlatform_InverseDynamic3_noBus.c:33917:1105: error: too many arguments to function call, expected 6, have 9
    if(!omc_StewartPlatform_Functions_validPose(threadData, omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1620] /* gp.base.D PARAM */), (data->simulationInfo->realParameter[1621] /* gp.base.De PARAM */), (data->simulationInfo->realParameter[1626] /* gp.base.thickness PARAM */), (data->simulationInfo->realParameter[1625] /* gp.base.mass PARAM */), (data->simulationInfo->realParameter[1622] /* gp.base.alpha PARAM */), (data->simulationInfo->realParameter[1623] /* gp.base.area PARAM */), (data->simulationInfo->realParameter[1627] /* gp.base.volume PARAM */), (data->simulationInfo->realParameter[1624] /* gp.base.density PARAM */)), omc_StewartPlatform_Types_DiscParameters(threadData, (data->simulationInfo->realParameter[1762] /* gp.platform.D PARAM */), (data->simulationInfo->realParameter[1763] /* gp.platform.De PARAM */), (data->simulationInfo->realParameter[1768] /* gp.platform.thickness PARAM */), (data->simulationInfo->realParameter[1767] /* gp.platform.mass PARAM */), (data->simulationInfo->realParameter[1764] /* gp.platform.alpha PARAM */), (data->simulationInfo->realParameter[1765] /* gp.platform.area PARAM */), (data->simulationInfo->realParameter[1769] /* gp.platform.volume PARAM */), (data->simulationInfo->realParameter[1766] /* gp.platform.density PARAM */)), tmp105, tmp106, tmp107, 1.798, 0.998))
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./StewartPlatform_InverseDynamic3_noBus_functions.h:336:38: note: 'omc_StewartPlatform_Types_DiscParameters' declared here
StewartPlatform_Types_DiscParameters omc_StewartPlatform_Types_DiscParameters (threadData_t *threadData, modelica_real omc_D, modelica_real omc_De, modelica_real omc_thickness, modelica_real omc_mass, modelica_real omc_alpha);
                                     ^
4 errors generated.
mingw32-make: *** [<builtin>: StewartPlatform_InverseDynamic3_noBus.o] Error 1
mingw32-make: *** Waiting for unfinished jobs....
mingw32-make: Leaving directory 'C:/Users/jackj/AppData/Local/Temp/OPENMO~1/OMEdit/STEWAR~1.STE'
Compilation process failed. Exited with code 0x00000002.

--- a/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
+++ b/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
@@ -428,7 +428,7 @@ template recordDeclaration(RecordDeclaration recDecl)
                       listLength(variables))%>
 
     <%recordConstructorDef(r.name, r.name, r.variables)%>
-    <%recordModelicaCallConstrctor(r.name, r.variables)%>
+    <%recordCreateFromVarsDef(r.name, r.variables)%>
 
     <%recordCopyDef(r.name, r.variables)%>
     <%if r.usedExternally then recordCopyExternalDefs(r.name, r.variables)%>
@@ -532,7 +532,7 @@ template recordDeclarationFullHeader(RecordDeclaration recDecl)
     let copy_to_vars_name_p = '<%copy_to_vars_name%>_p'
     let copy_to_vars_inputs = r.variables |> var as VARIABLE(__) => (", " + varType(var) + "* in_" + crefStr(var.name))
 
-    let modelica_ctor_name = 'omc_<%rec_name%>'
+    let create_from_vars_name = '<%rec_name%>_create_from_vars'
     let modelica_ctor_inputs = r.variables |> var as VARIABLE(__) => (", " + varType(var) + " in_" + crefStr(var.name))
 
       <<
@@ -575,9 +575,7 @@ template recordDeclarationFullHeader(RecordDeclaration recDecl)
         #define <%cpy_from_external_macro_name%>(src,dst) <%cpy_from_external_func_name%>(&src, &dst)
         >>
       %>
-      // This function should eventually replace the default 'modelica' record constructor funcition
-      // that omc used to generate, i.e., replace functionBodyRecordConstructor template.
-      // <%rec_name%> <%modelica_ctor_name%>(threadData_t *threadData <%modelica_ctor_inputs%>);
+      <%rec_name%> <%create_from_vars_name%>(threadData_t *threadData <%modelica_ctor_inputs%>);
 
       // This function is not needed anymore. If you want to know how a record
       // is 'assigned to' in simulation context see assignRhsExpToRecordCrefSimContext and
@@ -644,25 +642,16 @@ template recordCopyExternalDefs(String rec_name, list<Variable> variables)
   >>
 end recordCopyExternalDefs;
 
-template recordModelicaCallConstrctor(String rec_name, list<Variable> variables)
+template recordCreateFromVarsDef(String rec_name, list<Variable> variables)
  "Generates code for creating and initializing a record given values for
-  ALL its members. This is basically what we used to generate before. However
-  that one did not handle arrays and record record members properly. This
-  should replace that function. For now it is commented until I have time to
-  clean out uses of the other one and replace it with this.
+  ALL its members. This is used internally by the generated code to reconstruct
+  records from (the scattered) simulation member variables of the record. We do
+  this when we have to send a record used in equation context to a function.
 
   Note that this is defferent from the constructors we have. This is the function
   to be used when you do R(...). not for cases where you have
 
-  R r1; (uses default constructor, i e., recordDeclarationFullHeader and recordConstructorDef)
-  R r2(a=2); (uses extra constructor, i e., recordDeclarationExtraCtor and  recordConstructorDef)
-  ...
-
-  But for cases like these.
-  a := R() (should use this function)
-
-  The difference is this function creates and returns a NEW record instance. The others
-  accept an instance and initialize it (including any allocations needed by the MEMBERS).
+  This function creates and returns a NEW record instance.
   "
 ::=
   let &varCopies = buffer ""
@@ -674,10 +663,7 @@ template recordModelicaCallConstrctor(String rec_name, list<Variable> variables)
                             (", " + varType(var) + " " + src_pref + contextCrefNoPrevExp(var.name, contextFunction, &auxFunction))
                 )
   <<
-  // This function should eventually replace the default 'modelica' record constructor funcition
-  // that omc used to generate, i.e., replace functionBodyRecordConstructor template.
-  /*
-  <%rec_name%> omc_<%rec_name%>(threadData_t *threadData <%inputs%>) {
+  <%rec_name%> <%rec_name%>_create_from_vars(threadData_t *threadData <%inputs%>) {
     <%rec_name%> dst;
     // TODO Improve me. No need to initialize the record members with defaults in <%rec_name%>_construct
     // We should just do the allocs here and then copy the input parameters as default values instead.
@@ -685,9 +671,8 @@ template recordModelicaCallConstrctor(String rec_name, list<Variable> variables)
     <%varCopies%>
     return dst;
   }
-  */
   >>
-end recordModelicaCallConstrctor;
+end recordCreateFromVarsDef;
 
 template recordMemberCopy(Variable var, String src_pref, String dst_pref, Text &varCopies, Text &auxFunction)
   "Generates code for copying memembers of a record during a record copy operation.
@@ -5395,7 +5380,7 @@ template daeExpCrefRhsSimContext(Exp ecr, Context context, Text &preExp,
   case ecr as CREF(componentRef = cr, ty = t as T_COMPLEX(complexClassType = record_state, varLst = var_lst)) then
     let vars = var_lst |> v => (", " + constVarOrDaeExp(v, cr, context, &preExp, &varDecls, &auxFunction))
     let record_type_name = underscorePath(ClassInf.getStateName(record_state))
-    'omc_<%record_type_name%>(threadData<%vars%>)'
+    '<%record_type_name%>_create_from_vars(threadData<%vars%>)'
 
   case ecr as CREF(componentRef=cr, ty=T_ARRAY(ty=aty, dims=dims)) then
     let type = expTypeShort(aty)

--- a/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
+++ b/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
@@ -532,8 +532,10 @@ template recordDeclarationFullHeader(RecordDeclaration recDecl)
     let copy_to_vars_name_p = '<%copy_to_vars_name%>_p'
     let copy_to_vars_inputs = r.variables |> var as VARIABLE(__) => (", " + varType(var) + "* in_" + crefStr(var.name))
 
-    let create_from_vars_name = '<%rec_name%>_create_from_vars'
-    let modelica_ctor_inputs = r.variables |> var as VARIABLE(__) => (", " + varType(var) + " in_" + crefStr(var.name))
+    let wrap_vars_macro_name = '<%rec_name%>_wrap_vars'
+    let wrap_vars_func_name = '<%wrap_vars_macro_name%>_p'
+    let wrap_vars_macro_inputs = r.variables |> var as VARIABLE(__) => (", " + "in_" + crefStr(var.name))
+    let wrap_vars_func_inputs = r.variables |> var as VARIABLE(__) => (", " + varType(var) + " in_" + crefStr(var.name))
 
       <<
       <% match aliasName
@@ -575,7 +577,9 @@ template recordDeclarationFullHeader(RecordDeclaration recDecl)
         #define <%cpy_from_external_macro_name%>(src,dst) <%cpy_from_external_func_name%>(&src, &dst)
         >>
       %>
-      <%rec_name%> <%create_from_vars_name%>(threadData_t *threadData <%modelica_ctor_inputs%>);
+
+      void <%wrap_vars_func_name%>(threadData_t *threadData , void* v_dst <%wrap_vars_func_inputs%>);
+      #define <%wrap_vars_macro_name%>(td, dst <%wrap_vars_macro_inputs%>) <%wrap_vars_func_name%>(td, &dst <%wrap_vars_macro_inputs%>)
 
       // This function is not needed anymore. If you want to know how a record
       // is 'assigned to' in simulation context see assignRhsExpToRecordCrefSimContext and
@@ -643,33 +647,39 @@ template recordCopyExternalDefs(String rec_name, list<Variable> variables)
 end recordCopyExternalDefs;
 
 template recordCreateFromVarsDef(String rec_name, list<Variable> variables)
- "Generates code for creating and initializing a record given values for
+ "Generates code for creating and initializing (shallow copies) a record given values for
   ALL its members. This is used internally by the generated code to reconstruct
   records from (the scattered) simulation member variables of the record. We do
   this when we have to send a record used in equation context to a function.
 
-  Note that this is defferent from the constructors we have. This is the function
-  to be used when you do R(...). not for cases where you have
-
-  This function creates and returns a NEW record instance.
+  Note that this is defferent from the constructors we have. This one expects
+  all the memebers to be already created (allocated and given values whatever the value is).
+  Its job is to 'wrap' these variables by a given record instance, e.g., so it can be sent to
+  functions from simulation code.
   "
 ::=
-  let &varCopies = buffer ""
   let &auxFunction = buffer ""
-  let dst_pref = 'dst.'
-  let src_pref = ' in'
-  let _ = (variables |> var => recordMemberCopy(var, src_pref, dst_pref, &varCopies, &auxFunction) ;separator="\n")
-  let inputs = (variables |> var as VARIABLE(__) =>
+  let dst_pref = 'dst->'
+  let src_pref = 'in'
+
+  let varCopies = (variables |> var => match var
+    case var as VARIABLE(__) then
+      let dstName = dst_pref + contextCrefNoPrevExp(var.name, contextFunction, &auxFunction)
+      let srcName = src_pref + contextCrefNoPrevExp(var.name, contextFunction, &auxFunction)
+      '<%dstName%> = <%srcName%>;<%\n%>'
+    else error(sourceInfo(), "recordCreateFromVarsDef: Unhandled variable type"))
+
+  let fn_inputs = (variables |> var as VARIABLE(__) =>
                             (", " + varType(var) + " " + src_pref + contextCrefNoPrevExp(var.name, contextFunction, &auxFunction))
                 )
+
+  let ctor_additional_inputs = (variables |> var as VARIABLE(__) => if var.bind_from_outside
+                                  then (", " + "in_" + crefStr(var.name))
+                                  )
   <<
-  <%rec_name%> <%rec_name%>_create_from_vars(threadData_t *threadData <%inputs%>) {
-    <%rec_name%> dst;
-    // TODO Improve me. No need to initialize the record members with defaults in <%rec_name%>_construct
-    // We should just do the allocs here and then copy the input parameters as default values instead.
-    <%rec_name%>_construct(threadData, dst);
+  void <%rec_name%>_wrap_vars_p(threadData_t *threadData, void* v_dst <%fn_inputs%>) {
+    <%rec_name%>* dst = (<%rec_name%>*)(v_dst);
     <%varCopies%>
-    return dst;
   }
   >>
 end recordCreateFromVarsDef;
@@ -5380,7 +5390,9 @@ template daeExpCrefRhsSimContext(Exp ecr, Context context, Text &preExp,
   case ecr as CREF(componentRef = cr, ty = t as T_COMPLEX(complexClassType = record_state, varLst = var_lst)) then
     let vars = var_lst |> v => (", " + constVarOrDaeExp(v, cr, context, &preExp, &varDecls, &auxFunction))
     let record_type_name = underscorePath(ClassInf.getStateName(record_state))
-    '<%record_type_name%>_create_from_vars(threadData<%vars%>)'
+    let tmpRec = tempDecl(record_type_name, &varDecls)
+    let &preExp += '<%record_type_name%>_wrap_vars(threadData,<%tmpRec%><%vars%>);<%\n%>'
+    '<%tmpRec%>'
 
   case ecr as CREF(componentRef=cr, ty=T_ARRAY(ty=aty, dims=dims)) then
     let type = expTypeShort(aty)

